server/http_tunnel.go

```diff
@@ -27,9 +28,19 @@ func StartHTTPSTunnel(addr, externalProxy, certFile, keyFile, username, password
 	proxy := goproxy.NewProxyHttpServer()
 
 	if externalProxy != "" {
-		proxy.Tr = &http.Transport{Proxy: func(req *http.Request) (*url.URL, error) {
-			return url.Parse(externalProxy)
-		}}
+		u, err := url.Parse(externalProxy)
+		if err != nil {
+			log.Printf("HTTP(S) Tunnel: failed to parse external proxy: %s\n", err.Error())
+			return
+		}
+		proxy.Tr = &http.Transport{
+			Proxy: func(req *http.Request) (*url.URL, error) {
+				return u, nil
+			},
+			TLSClientConfig: &tls.Config{
+				ServerName: u.Hostname(),
```

Comment:
Do we need to set this up when StartHTTPTunnel() is being called?

Reply:
Yes. StartHTTPTunnel() and StartHTTPSTunnel() differ on whether we listen on TLS or plain HTTP. Both of them can have an external TLS proxy, so they will both need this.